functions/ask.js

```
@@ -0,0 +1,155 @@
+const { google } = require('googleapis')
+const axios = require('axios')
+const { dockStart } = require('@nlpjs/basic');
+
+const {
+    GOOGLE_CLIENT_EMAIL,
+    GOOGLE_PRIVATE_KEY
+} = process.env
+
+const privateKey = GOOGLE_PRIVATE_KEY.replace(/\\n/gm, '\n')
+const treeDataUrl = 'https://graduate-16c74.firebaseio.com/tree.json/'
+
+const scopes = [
+    "https://www.googleapis.com/auth/userinfo.email",
+    "https://www.googleapis.com/auth/firebase.database"
+];
+const jwt = new google.auth.JWT(GOOGLE_CLIENT_EMAIL, null, privateKey, scopes);
+
+exports.handler = async function (event, context) {
+    const receivedMessage = event.queryStringParameters.message
+    const receivedQuickAnswer = event.queryStringParameters.quickAnswer
+
+    try {
+
+        const token = await new Promise((resolve, reject) => jwt.authorize(async (error, tokens) => {
+            if (error) {
+                reject(new Error('Error making request to generate token'))
+            } else if (!tokens.access_token) {
+                reject(new Error('Provided service account doest not have permission to generate access tokens'))
+            }
+            resolve(tokens.access_token)
+        }))
+
+        if (!token)  return null;
```

No token is error right? I think that you should also return a error in the response. Again i would suggest you to look at [middy](https://middy.js.org/) for this.